cilantro_audit/create_completed_audit_page.py

```diff
@@ -0,0 +1,131 @@
+
+    # put all questions on the screen for the auditor to respond to
+    def populate_audit(self, audit_name):
+        target = audit_name
+        try:
+            template = AuditTemplate.objects().filter(title=target).first()  # for now, while there can be duplicates
+        except AttributeError:
+            # TO DO - SOMETHING
+            pass
+
+        self.audit_title = template.title
+        for question in template.questions:
+            self.stack_list.height += 200
+            a_temp = AnswerModule()
+            a_temp.question = question
+            a_temp.question_text = question.text
+            self.stack_list.add_widget(a_temp)
+            self.questions.append(a_temp)
+
+    # Return the associated severity with question's response
+    def question_severity(self, question):
+        if question.response == Response.yes():
+            return question.question.yes
+        elif question.response == Response.no():
+            return question.question.no
+        return question.question.other
+
+    # Function called after user selects yes on the confirmation popup
+    def submit_audit(self, callback):
+        completed_audit = CompletedAuditBuilder()
+        completed_audit.with_title(self.audit_title)
+        # The object returned from the .kv is a TextField, with a member text
+        completed_audit.with_auditor(self.auditor_name.text)
+        for a in self.questions:
+            if a.other_comments.text:
+                temp_answer = Answer(text=a.question.text, severity=self.question_severity(a), response=a.response,
+                                     comment=a.other_comments.text)
+            else:
+                temp_answer = Answer(text=a.question.text, severity=self.question_severity(a), response=a.response)
+            completed_audit.with_answer(temp_answer)
+
+        completed_audit.build().save()
+
+    def submit_audit_pop(self, manager):
+        error_message = self.check_audit()
+        if error_message != "":
+            show = ErrorPop()
+            show.error_message.text = error_message
+        else:
+            show = ConfirmationPop()
+            show.yes.bind(on_press=self.clear_page)
+            show.yes.bind(on_press=self.submit_audit)
+        show.manager = manager
+        show.open()
+
+    # On press method for the back button
+    def back(self, manager):
+        show = ConfirmationPop()
+        show.yes.bind(on_press=self.clear_page)
+        show.manager = manager
+        show.open()
+
+    # Empties stack list and question list, should enable leaving early without a problem...
+    def clear_page(self, callback):
+        self.audit_title = ""
+        for question in self.questions:
+            self.stack_list.remove_widget(question)
+            self.stack_list.height -= 200
+        self.questions.clear()
+
+    # Ensures that the completedAudit has everything filled out before trying to .save() it
+    def check_audit(self):
```

Comment:
I don't think the most recent version of Sean's branch is in here yet. At the sponsor review Sean showed that this would ensure comments were entered if other is selected. As it stands, this will throw a backend ValidationError and shut out the app without any popup warning.